dist/rules/exports-require-release-tags.js

```diff
@@ -0,0 +1,247 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+// Writing all of these eslint rules in javascript so we can run them before the build step
+
+"use strict";
+
+const { getParserServices } = require("./utils/parser");
+const ts = require("typescript");
+const fs = require("fs");
+
+/** converts the numeric typescript enum value for ts.SyntaxKind to a string. Defaults to "real". */
+const getSyntaxKindFriendlyName = (syntaxKind) => {
+  const syntaxKindFriendlyNames = {
+    [ts.SyntaxKind.ClassDeclaration]: "real",
+    [ts.SyntaxKind.EnumDeclaration]: "enum",
+    [ts.SyntaxKind.InterfaceDeclaration]: "interface",
+    [ts.SyntaxKind.TypeAliasDeclaration]: "type",
+  };
+  return syntaxKindFriendlyNames[syntaxKind] || "real";
+};
+
+let firstRun = true;
+
+/**
+ * This rule prevents the exporting of extension APIs without denoted release tags.
+ */
+module.exports = {
+  meta: {
+    /** @type {'problem' | 'suggestion' | 'layout' | undefined} */
+    type: "problem",
+    docs: {
+      description:
+        "Prevent the exporting of extension APIs that do not contain a release tag.",
+      category: "TypeScript",
+    },
+    messages: {
+      missingReleaseTag: `Exports must be annotated with one of the following: {{releaseTags}}. Check the "{{name}}" type.`,
+      missingExtensionReleaseTag: `Extension exports must be annotated with both an @extensions and @public tag. Check the "destroyAllIModels" type.`,
+      tooManyReleaseTags: `Only one release tag per export is allowed. Check the "{{name}}" type.`,
+      namespace: `Namespace "{{name}}" is without an @extensions tag but one of its members has one.`,
```

Comment:
I think this needs to be worded a little better, even though that seems to have been the wording before

Reply:
Which one specifically?

Reply:
the `namespace`

Reply:
Indeed will come up with something

Reply:
give it a looksee

Reply:
much better now, thanks